feat: convert SignPosts and LanguageList shortcodes to Nunjunks templates

mamieorine opened the pull request:
Fixes #9775 

Since both `{% SignPosts %}` and `{% LanguageList %}` shortcode which are only used from Nunjunks templates, this should instead be refactored to a Nunjunks template and imported to the content using {% include %}.

Changes proposed in this pull request:

- Remove `{% SignPosts %}` shortcode from the eleventy config
- Remove `{% LanguageList %}` shortcode from the eleventy config
- Convert SignPosts shortcode to a simple a Nunjunks template
- Convert LanguageList shortcode to a Nunjunks template 

For converting the LanguageList shortcode to Nunjunks, this component needs a javascript file that has logic for finding the available language for each post content. Thus, I kept the `LanguageList.js `and imported it into `content.11tydata.js.`. Please let me know if you have any suggestions on this, thank you.

matthiasrohmer commented on src/site/_includes/codelab.njk:
```diff
@@ -29,7 +29,8 @@ pageScripts:
         {% endif %}
 
-        {% SignPosts page.fileSlug %}
+        {% include 'partials/sing-post.njk' %}
```
New keyboard? 😄  `sign`.

matthiasrohmer reviewed (commented):
Bunch of typos. My suggestion is to remove the remaining needed logic to a filter instead and then do something like `page | languageList` in the template. Make sure to also move the `LanguageList` file out of the components directory then.

mamieorine commented:
@matthiasrohmer  I have fixed all typos and changed the way to enable using of functions in the Nunjunks template by implementing the supportedLanguages filter instead.

- Multiple languages of the content available
   https://deploy-preview-9849--web-dev-staging.netlify.app/defining-core-web-vitals-thresholds/

- Only English language available
   https://deploy-preview-9849--web-dev-staging.netlify.app/push-notifications-in-all-modern-browsers/

matthiasrohmer commented on src/site/_filters/language-list.js:
```diff
@@ -15,52 +15,55 @@
+ *
+ * @param {String} lang The current lang of a post to look up.
+ * @param {EleventyCollectionItem} post The post to look up.
+ * @return {String[][]} The list of supported languages for a specific post
```
Question: is that a valid type?

mamieorine replied:
Yes, this function will return the array of supported languages like:

```
[
  [ 'en', '/i18n/en/webrtc-standard-announcement/', 'English' ],
  [ 'es', '/i18n/es/webrtc-standard-announcement/', 'Español' ],
  [ 'pt', '/i18n/pt/webrtc-standard-announcement/', 'Português' ],
  [ 'ru', '/i18n/ru/webrtc-standard-announcement/', 'Русский' ],
  [ 'zh', '/i18n/zh/webrtc-standard-announcement/', '中文' ],
  [ 'ja', '/i18n/ja/webrtc-standard-announcement/', '日本語' ],
  [ 'ko', '/i18n/ko/webrtc-standard-announcement/', '한국어' ]
]
```

matthiasrohmer commented on src/site/_filters/language-list.js:
```diff
@@ -15,52 +15,55 @@
-  if (langHrefs.length === 0) {
-    return '';
-  }
+  if (languages.length === 0) return;
```
That essentially is `return undefined` which is not a valid return type according to the JSDoc. This should probably be `return []`.

matthiasrohmer commented on src/site/_filters/language-list.js:
```diff
@@ -15,52 +15,55 @@
 
-module.exports = (url, lang) => {
-  const langHrefs = getTranslatedUrls(url).filter(
+const supportedlanguages = (post, lang) => {
```
Nit: camel-case, so `supportedLanguages`.

matthiasrohmer commented on src/site/_includes/partials/language-list.njk:
```diff
@@ -0,0 +1,19 @@
+
+      <span>
+        <a href={{langHref}} class="post-signpost__link" translate="no" lang={{lang}}>{{langName}}</a>
+        {%- if loop.index !== numberOfLanguages -%}, {%- endif -%}
+        {% if loop.index === (numberOfLanguages - 1) %} and {% endif %}
```
I think you don't need `numberOfLanguages` here and the loop can be simplified, as you have `loop.first`, `loop.last` and `loop.length` in Nunjucks. See https://mozilla.github.io/nunjucks/templating.html#:~:text=loop.first%3A%20boolean,loop.last

matthiasrohmer commented on src/site/_includes/partials/sign-post.njk:
```diff
@@ -0,0 +1,23 @@
+
+        <a href={{postHref}} class="post-signpost__link">{{paths[pathName].title | i18n(locale)}}</a>
+
+        {% if loop.index !== numberOfPaths %}
```
Use `loop.last` instead.
